sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -648,29 +648,41 @@ def convert_to_placeholder(input_value: str) -> str:
             container_spec.resources.resource_cpu_request = (
                 convert_to_placeholder(
                     task.container_spec.resources.cpu_request))
+            container_spec.resources.cpu_request = compiler_utils.validate_cpu_request_limit_to_float(
+                cpu=convert_to_placeholder(
+                    task.container_spec.resources.cpu_request))
         if task.container_spec.resources.cpu_limit is not None:
             container_spec.resources.resource_cpu_limit = (
                 convert_to_placeholder(task.container_spec.resources.cpu_limit))
+            container_spec.resources.cpu_limit = compiler_utils.validate_cpu_request_limit_to_float(
+                cpu=convert_to_placeholder(
+                    task.container_spec.resources.cpu_limit))
         if task.container_spec.resources.memory_request is not None:
             container_spec.resources.resource_memory_request = (
                 convert_to_placeholder(
                     task.container_spec.resources.memory_request))
+            container_spec.resources.memory_request = compiler_utils.validate_memory_request_limit_to_float(
+                memory=convert_to_placeholder(
+                    task.container_spec.resources.memory_request))
         if task.container_spec.resources.memory_limit is not None:
             container_spec.resources.resource_memory_limit = (
                 convert_to_placeholder(
                     task.container_spec.resources.memory_limit))
+            container_spec.resources.memory_limit = compiler_utils.validate_memory_request_limit_to_float(
+                memory=convert_to_placeholder(
+                    task.container_spec.resources.memory_limit))
```

Comment:
I think this is getting a little messy. I would prefer it if we kept the logic of how we validate the deprecated fields similar to how it used to be, and keep it in `class PipelineTask:` the way I suggest doing this is: 

Return [these](https://github.com/kubeflow/pipelines/pull/11097/files#diff-e87964fa6e1275bbe5b60d8cb7446aeb5854514987053d1c875092129b0975e0L242-L247) fields as float types, but rename them as: 

```python
    cpu_request_deprecated: Optional[float] = None
    cpu_limit_deprecated: Optional[float] = None
    memory_request_deprecated: Optional[float] = None
    memory_limit_deprecated: Optional[float] = None
```

Then move `validate_cpu_request_limit_to_float` & `validate_memory_request_limit_to_float` back in to `pipeline_task.py`, and when  `set_cpu_request`, `set_cpu_limit`, `set_memory_request`, `set_memory_limit`  is called, we set the deprecated fields the same as we did before the breaking change, and validate right before assignment. This keeps the logic functioning similarly as it used to, and it keeps all the logic close to gether so it's easier to remove it when we sunset these fields.

Reply:
I only followed the way it's done for Accelerator fields. So Accelerator should be changed as well?